prefect_sqlalchemy/credentials.py

```diff
@@ -0,0 +1,98 @@
+"""Credential classes used to perform authenticated interactions with SQLAlchemy"""
+
+from dataclasses import dataclass
+from typing import TYPE_CHECKING, Any, Dict, Optional
+
+from sqlalchemy.engine.url import URL
+from sqlalchemy.ext.asyncio import create_async_engine
+
+if TYPE_CHECKING:
+    from sqlalchemy.ext.asyncio.engine import AsyncConnection
+
+
+@dataclass
+class SQLAlchemyCredentials:
+    """
+    Dataclass used to manage authentication with SQLAlchemy.
+    The engine should be disposed manually at the end of the flow.
+
+    Args:
+        driver: The driver name, e.g. "postgresql+asyncpg"
+        database: The name of the database to use.
+        user: The user name used to authenticate.
+        password: The password used to authenticate.
+        host: The host address of the database.
+        port: The port to connect to the database.
+        query: A dictionary of string keys to string values to be passed to
+            the dialect and/or the DBAPI upon connect. To specify non-string
+            parameters to a Python DBAPI directly, use connect_args.
+        connect_args: The options which will be passed
+            directly to the DBAPI's connect() method as
+            additional keyword arguments.
+    """
+
+    driver: str
+    user: str
+    password: str
+    database: str
+    host: Optional[str] = "localhost"
+    port: Optional[str] = 5432
+    query: Optional[Dict[str, str]] = None
+    connect_args: Optional[Dict[str, Any]] = None
+
+    def __post_init__(self):
+        """
+        Initializes the engine.
+        """
+        async_drivers = {
+            "postgresql": "postgresql+asyncpg",
+            "sqlite": "sqlite+aiosqlite",
+            "mysql": "mysql+aiomysql",
+        }  # replace with async drivers
+        drivername = async_drivers.get(self.driver.lower(), self.driver)
+
+        url = URL.create(
+            drivername=drivername,
+            username=self.user,
+            password=self.password,
+            database=self.database,
+            host=self.host,
+            port=self.port,
+            query=self.query,
+        )
+        connect_args = self.connect_args or {}
+        self.engine = create_async_engine(url, connect_args=connect_args)
```

Comment:
What happens if a user passes in a driver that isn't in the `async_drivers` list? Are we able to create an async engine for a MSSQL or Oracle database?

Reply:
Not at the moment: https://github.com/sqlalchemy/sqlalchemy/issues/6521

But at the same time, if it gets released without us being aware, I don't want users to be blocked from using it.

Reply:
Another thing is that each dialect has multiple async drivers too:

postgresql:
`postgresql+psycopg_async`
`postgresql+asyncpg`

mysql:
`mysql+asyncmy`
`mysql+aiomysql`

I am not sure if there's a programmatic way of retrieving these, I did so by searching Github
https://github.com/sqlalchemy/sqlalchemy/search?p=2&q=async+%3D+True

Reply:
Based on my exploration, non-async engines aren't able to be created with `create_async_engine`. This may be a rare case where we want a sync and an async task unless we can find a way to abstract sync vs async engines from the user.

Reply:
If we do support both, how do we let them know that they are using the async/sync version? Perhaps a logger message that says "sync" vs "async"?

Reply:
They both have the `.execute` method, but the will work differently for sync vs async use cases. The async engine will return coroutines on method calls while the sync engine wouldn't. 

I think we'll need different tasks for the sync and async engines primarily because we can't support SQL Server and Oracle with just an async task due to the lack of async support for those drivers in SQLAlchemy. We can add a sync task in a separate PR since this one is on the cusp of being merged.